### DISCUSSION TITLE
Is it currently possible to connect with sMQTTbroker over paho-mqtt.js

### DISCUSSION THREAD

**Discussion by RobertHolscher** [terrorsl/sMQTTBroker#47] [Q&A] [answered]
I have webapplications using paho-mqtt.js to connect to MQTT brokers. However is it only working over websockets. Is it possible to use paho-mqtt.js functions to connect to sMQTTbroker ? I only see non websocket connections, and websocket is the only connection which is available in Javascript. Sofar i was not succesful. Maybe any ideas or maybe a feature request ?

**Comment by terrorsl** [comment, marked as answer]
experimental version here [websocket version](https://github.com/terrorsl/sMQTTBroker/tree/websocket)

**Comment by Hai-Elec** [comment]
In case you need more about paho eclipse with sMQTT on android app. I use android studio and load paho eclipse mqtt to app folder, and AI assistant of android studio help me write the code to connect to sMQTT on ESP8266. Thanks for your library, [terrorsl](https://github.com/terrorsl)
I think use app is better web applicant because ESP8266 usually is used for small project, and if we need to store data, need a server. So, app on mobile phone can store data, and it is more convenient than web applicant to store some small data. 
I have 2 project with web-applicant on ESP8266 already, I use XAMPP server which installed on my PC to log data. ESP8266 use HTML command.

**Reply by terrorsl** [reply]
Hi, I would use the esp8266 as a broker and the phone and computer storage as clients.

**Comment by Hai-Elec** [comment]
I try to send data from Broker back to my phone, 
After confuse check publish function in sMQTTBroker.cpp, I ask the AI assistant in android studio and it help me to write the code to publish data from broker to my phone app and it worked. 

```
#include <sys/times.h>

extern "C" {
  int _times_r(struct _reent *r, struct tms *t) {
    t->tms_utime = 0;
    t->tms_stime = 0;
    t->tms_cutime = 0;
    t->tms_cstime = 0;
    return 0;
  }
}

// long coding here................................


MyBroker broker;

void sensor_data() //get ADC pin value
{
  double tempget = Thermister(AnalogRead());
  
  if (tempget>80){//faultytempalarm = 1;
  //analogWriteFreq(4350);}//buzzer for faulty temp
  }
  else{
   // faultytempalarm = 0;
  }
   String sensor_value = String(tempget); // Convert the double to an Arduino String

  // Use the public, high-level publish function.
  // It requires a C-style string for the payload, so we use .c_str() to convert.
  broker.publish("temp", sensor_value.c_str());

};
```

**Reply by terrorsl** [reply]
I don't like that AI bots replace the timer function. I recommend doing it like here(#31).
